torch/jit/frontend.py

```diff
@@ -291,6 +292,20 @@ def _forward(self):
 
     return build_def(ctx, fn_def, type_line, def_name, self_name=self_name)
 
+def is_torch_jit_ignore_context_manager(stmt):
+    # checks if the statement is torch.jit.ignore context manager
+    if isinstance(stmt.items[0].context_expr, ast.Call):
+        # extract torch part
+        function = stmt.items[0].context_expr.func
+        if isinstance(function, ast.Attribute):
+            attr_name = function.attr
+            attr_value = function.value
+            if attr_name == "ignore" and isinstance(attr_value, ast.Attribute):
+                # there should be at most two nested attributes (e.g torch.jit.ignore)
+                if attr_value.attr == "jit" and isinstance(attr_value.value, ast.Name):
+                    if attr_value.value.id == "torch":
+                        return True
+    return False
```

Comment:
It is OK to fix this in next PR:

This is a bit fragile. It will match the common case of `with torch.jit.ignore`. But it would fail for following cases:

```
cm = torch.jit.ignore(...)
with cm:
    ......
```

```
import torch as th
with th.jit.ignore(...):
   ......
```

```
from torch.jit import ignore
with ignore(...):
    ......
```

I think we can try plumb a resolution callback to here in order to correctly parse `ignore`.